spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java

```diff
@@ -225,6 +225,8 @@ public void register(@PathVariable("type") ApplicationType type, @PathVariable("
 			@RequestParam("uri") String uri, @RequestParam(name = "metadata-uri", required = false) String metadataUri,
 			@RequestParam(value = "force", defaultValue = "false") boolean force) {
 
+		if (!isAppNameValid(name))
+			throw new IllegalArgumentException("Application name: " + name + " cannot contain ':'");
```

Comment:
This `isAppNameValId` if gets used for other wildcard validations, then we may fall short on the error message - like cannot contain  `:`.
Maybe declare a wildcard constant and check with contains with the same constant.

Reply:
That makes sense, currently, I'm checking exactly what all validation would the project owners want, will update this accordingly.

Reply:
Updated the validation method.